src/momento/internal/aio/_vector_index_data_client.py

```diff
@@ -136,6 +139,53 @@ async def search(
             self._log_request_error("search", e)
             return Search.Error(convert_error(e, Service.INDEX))
 
+    async def search_and_fetch_vectors(
+        self,
+        index_name: str,
+        query_vector: list[float],
+        top_k: int,
+        metadata_fields: Optional[list[str]] | AllMetadata = None,
+        score_threshold: Optional[float] = None,
+    ) -> SearchAndFetchVectorsResponse:
+        try:
+            self._log_issuing_request("SearchAndFetchVectors", {"index_name": index_name})
+            _validate_index_name(index_name)
+            _validate_top_k(top_k)
+
+            query_vector_pb = vectorindex_pb._Vector(elements=query_vector)
+            if isinstance(metadata_fields, AllMetadata):
+                metadata_fields_pb = vectorindex_pb._MetadataRequest(all=vectorindex_pb._MetadataRequest.All())
+            else:
+                metadata_fields_pb = vectorindex_pb._MetadataRequest(
+                    some=vectorindex_pb._MetadataRequest.Some(
+                        fields=metadata_fields if metadata_fields is not None else []
+                    )
+                )
+
+            no_score_threshold = None
+            if score_threshold is None:
+                no_score_threshold = vectorindex_pb._NoScoreThreshold()
```

Comment:
Not 100% on what significance `no_score_threshold` has here, why initialized as `None` and then asigned an empty `vectorindex_pb._NoScoreThreshold()` if the `score_threshold` is `None`

Reply:
It looks like either score_threshold or no_score_threshold must be set in the request object. If score_threshold is None, no_score_threshold is set, otherwise it is left as None. Then both are added to the request object.

Reply:
Protobuf in python forces you to initialize `oneof` fields in the message constructor. In our case there either is a score threshold or there isn't.

If there is a score threshold, then we need to set `score_threshold` to that value and `no_score_threshold` to `None` (indicating it is not part of the oneof).

If there isn't a score threshold, then we set `score_threshold` to `None` and we instantiate `no_score_threshold`.